llvm/lib/Transforms/IPO/Attributor.cpp

```diff
@@ -1749,6 +1749,10 @@ bool Attributor::checkForAllCallees(
   return Pred(Callees.getArrayRef());
 }
 
+bool noRevisitUser(const User *user) {
+  return isa<PHINode>(user) || !isa<Instruction>(user);
```

Comment:
I'd recommend to prevent adding a phi or instruction on the worklist to begin with. 
The AddUsers function can have this check before the User is pushed on the worklist. 
For instance these checks can go here: https://github.com/llvm/llvm-project/blob/530ec6d11462f7801c9cfc819b0deda876ca3d2d/llvm/lib/Transforms/IPO/Attributor.cpp#L1784

Reply:
I tried this locally, and it seems to break a lot of tests. Not sure exactly why this fails, but I suspect it might be caused by the `checkForAllCallSites` call also using the `AddUsers` closure. I think we should just keep the check in the worklist loop for now.

```cpp
if (!noRevisitUser(UU.getUser()))
  Worklist.push_back(&UU);
```